mbed_app.json5

```diff
@@ -51,7 +51,16 @@
         //     "mcuboot.read-granularity": 1,
         //     "qspif.QSPI_MIN_PROG_SIZE": 1
         // },
-
+        "DISCO_F746NG": {
+            "target.restrict_size": "0x18000",
```

Comment:
Could we get some comments here?  I upgraded these files to be json5 now so don't hold back on the detail.  Like, where do these sizes and addresses come from?

Reply:
The considerations are mostly to maximize space for primary application. If QSPIF/SPIF is used as secondary block device, the bootloader will need more than 48KB. So I give it 64KB. Then the last sector for scratch. I will try to add comments into the json5 file. I still mostly use JSON files and so haven't got into the habit of adding comments yet.

Reply:
Believe me it's very freeing after all these years of not being able to have comments